src/climate_data/generate/scenario_daily.py

```
@@ -173,113 +177,72 @@ def generate_scenario_daily_main(  # noqa: PLR0912, PLR0915, C901
-
-    ensemble_anomaly = xr.Dataset()
-    for i, (source, source_anomalies) in enumerate(anomalies.items()):
-        sid = f"Source {i+1}/{len(source_paths)}: {source}"
-        print(f"Downscaling {i+1}/{len(anomalies)}: {source}")
-
-        source_ensemble_anomaly = xr.Dataset()
-        total_count = 0
-        for j, (res, (count, v_anomaly)) in enumerate(source_anomalies.items()):
-            res_id = f"{sid}, Resolution {j} / {len(source_anomalies)}: {res}"
-            print(f"Downscaling {res_id}")
-
-            if source_ensemble_anomaly.nbytes:
-                source_ensemble_anomaly += utils.interpolate_to_target_latlon(
-                    v_anomaly, method="linear"
-                )
-            else:
-                source_ensemble_anomaly = utils.interpolate_to_target_latlon(
-                    v_anomaly, method="linear"
-                )
-            total_count += count
-        source_ensemble_anomaly /= total_count
-
-        if ensemble_anomaly.nbytes:
-            ensemble_anomaly += source_ensemble_anomaly
-        else:
-            ensemble_anomaly = source_ensemble_anomaly
-
-    ensemble_anomaly /= len(anomalies)
-
-    print("Computing scenario data")
+    # randomly select source, variant, load reference and target,
+    # compute anomaly, resample anomaly and compute scenario data
+    source_key = random.choice(list(source_paths.keys()))
+    variant_paths = random.choice(source_paths[source_key])
+    s_variant = f"{variant_paths[0].stem.split('_')[-1]}"
+    vid = f"{source_key}, Variant : {s_variant}"
+    # load reference (monthly) and target (daily for a given year)
+    try:
+        print(f"{vid}: Loading reference")
+        sref = transform(*[load_variable(vp, "reference") for vp in variant_paths])
+        print(f"{vid}: Loading target")
+        target = transform(*[load_variable(vp, year) for vp in variant_paths])
+
+    except KeyError:
+        print(f"{vid}: Bad formatting, skipping...")
```

I think since we're not outputting an ensemble, we just want to raise the error.  I.e. ditch the try-except block and just keep 188-191.  We do not want to start a workflow to produce 100 draws and have it complete without an error, but still produce fewer than 100 draws.